socs/agents/galildmc_axis_controller/agent.py

```diff
@@ -0,0 +1,1170 @@
+        """set_gearing_ratio(order)
+
+        **Task** - Set electronic gearing ratio between leader and follower axes.
+
+        Parameters:
+            order (str): Comma-separated string of gearing ratios for each axis,
+            similar in format as `set_gearing` task.  Example: ',,,,,1' sets a ratio
+            of 1 between the F follower and E leader axes, meaning F follows
+            E at the same speed.
+
+        """
+        order = params['order']
+        with self.lock.acquire_timeout(timeout=5, job='set_gearing_ratio') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            self.stage.set_gearing_ratio(order)
+
+        return True, f'Gearing ratio set to {order}'
+
+    @ocs_agent.param('axis', type=str)
+    def get_gearing_ratio(self, session, params):
+        """get_gearing_ratio(axis)
+
+        **Task** - Query the current electronic gearing ratio for a specified axis
+
+        Parameters:
+            axis (str): Axis to query (e.g., 'A')
+
+        """
+        axis = params['axis']
+        with self.lock.acquire_timeout(timeout=5, job='get_gearing_ratio') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            resp = self.stage.get_gearing_ratio(axis=axis)
+
+        return True, f'Gearing ratio for axis {axis} is {resp}'
+
+    @ocs_agent.param('configfile', type=str, default=None)
+    def input_configfile(self, session, params=None):
+        """input_configfile(configfile=None)
+
+        **Task** Upload GalilDMC Axis Controller configuration file to initialize device
+        and axes on device
+
+        Parameters:
+            configfile (str, optional):
+                name of .yaml config file. Defaults to the file set in the site config.
+
+
+        """
+        configfile = params['configfile']
+        if configfile is None:
+            configfile = self.configfile
+        if configfile is None:
+            raise ValueError("No configfile specified")
+        configfile = os.path.join(os.environ['OCS_CONFIG_DIR'], configfile)
+
+        with self.lock.acquire_timeout(timeout=5, job='input_configfile') as acquired:
+            if not acquired:
+                self.log.warn(f"Could not start Task because "
+                              f"{self.lock.job} is already running")
+                return False, "Could not acquire lock"
+
+            cfg = read_config(configfile)
+            try:
+                gal = cfg.get('galil', {})
+                self.cfg = cfg
+                self.motorsettings = gal.get('motorsettings', {})
+                self.axis_map = gal.get('motorconfigparams', {})
```

Comment:
If these keys are checked in `read_config` then you can just directly reference them instead of using `.get()`.